rubix-presto/src/main/java/com/qubole/rubix/presto/PrestoClusterManager.java

```diff
@@ -165,35 +208,49 @@ public boolean isMaster()
     @Override
     public List<String> getNodes()
     {
-        return nodesSupplier.get();
+        //return nodesSupplier.get();
+        try {
+            return nodesCache.get("nodeList");
+        }
+        catch (ExecutionException e) {
+            e.printStackTrace();
+        }
+        return null;
```

Comment:
we allow returning null but we do handle null in callers. There are two callers for this method, add null check in them